grapple/helpers.py

```diff
@@ -17,22 +17,28 @@
 field_middlewares = {}
 
 
-def register_streamfield_block(cls):
-    base_block = None
-    for block_class in inspect.getmro(cls):
-        if block_class in registry.streamfield_blocks:
-            base_block = registry.streamfield_blocks[block_class]
-
-    streamfield_types.append(
-        {
-            "cls": cls,
-            "type_prefix": "",
-            "interface": StreamFieldInterface,
-            "base_type": base_block,
-        }
-    )
-
-    return cls
+def register_streamfield_block(cls=None, extra_interfaces=()):
+    def decorator(cls):
+        base_block = None
+        for block_class in inspect.getmro(cls):
+            if block_class in registry.streamfield_blocks:
+                base_block = registry.streamfield_blocks[block_class]
+
+        streamfield_types.append(
+            {
+                "cls": cls,
+                "type_prefix": "",
+                "interfaces": (StreamFieldInterface, *extra_interfaces),
```

Comment:
I am torn between calling `extra_interfaces` just `interfaces`.

Also, at the moment if you pass `(StreamFieldInterface, MyCustomInterface)`, the whole GraphQL endpoint crashes with `Type TheStreamFieldBlock can only implement StreamFieldInterface once."`. Let's be more defensive

Reply:
Renamed to `interfaces=`, it's what I was initially using. And the `interfaces` tuple is now constructed from a set so that `StreamFieldInterface` gets deduplicated.